source/uri/Query.ooc

```diff
@@ -0,0 +1,99 @@
+Query: class {
+	_attributes: VectorList<Text>
+	_values: VectorList<Text>
+	attributes: VectorList<Text> {
+		get {
+			result := VectorList<Text> new()
+			for (i in 0 .. this _attributes count) {
+				result add(this _attributes[i] take())
+			}
+			result
+		}
+	}
+	values: VectorList<Text> {
+		get {
+			result := VectorList<Text> new()
+			for (i in 0 .. this _values count) {
+				result add(this _values[i] take())
+			}
+			result
+		}
+	}
+	init: func(=_attributes, =_values)
+	free: override func {
+		for (i in 0 .. this _attributes count)
+			this _attributes[i] free(Owner Receiver)
+		for (i in 0 .. this _values count)
+			this _values[i] free(Owner Receiver)
+		this _attributes free()
+		this _values free()
+		super()
+	}
+	toText: func -> Text {
+		result := Text empty
+		attributes := this attributes
+		values := this values
+		for (i in 0 .. attributes count) {
+			argument := attributes[i]
+			if (values[i] != Text empty)
+				argument += t"=" + values[i]
+			result += i == 0 ? argument : t";" + argument
+		}
+		if (attributes != null)
+			attributes free()
+		if (values != null)
+			values free()
+		result
+	}
+	contains: func(attribute: Text) -> Bool {
+		result := false
+		attributes := this attributes
+		for (i in 0 .. attributes count) {
+			if (attributes[i] == attribute) {
+				result = true
+				i = attributes count
+			}
+		}
+		if (attributes != null)
+			attributes free()
+		result
+	}
+	getValue: func(attribute: Text) -> Text {
+		result := Text empty
+		attributes := this attributes
+		values := this values
+		for (i in 0 .. attributes count) {
+			if (attributes[i] == attribute) {
+				result = values[i] take()
```

Comment:
maybe check if `attributes` and `values` have the same count somewhere, and throw exception if they dont